File: data/transcoder_evaluation_gfg/go/FIND_SUM_MODULO_K_FIRST_N_NATURAL_NUMBER_1.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(N int, K int) int {
	var (
		ans int = 0
		y   int = N / K
		x   int = N % K
	)
	ans = (K*(K-1)/2)*y + (x*(x+1))/2
	return ans
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{40, 46, 97, 63, 92, 60, 67, 61, 74, 67}
		param1    []int = []int{90, 64, 20, 1, 52, 35, 40, 62, 61, 41}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i], param1[i]) == f_gold(param0[i], param1[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/FIND_SUM_MODULO_K_FIRST_N_NATURAL_NUMBER_1.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut N: i32, mut K: i32)
 -> i32 {
    let mut ans: i32 = 0 as i32;
    let mut y: i32 = N / K;
    let mut x: i32 = N % K;
    ans =
        K * (K - 1 as i32) / 2 as i32 * y +
            x * (x + 1 as i32) / 2 as i32;
    return ans;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [40 as i32, 46 as i32, 97 as i32,
         63 as i32, 92 as i32, 60 as i32,
         67 as i32, 61 as i32, 74 as i32,
         67 as i32];
    let mut param1: [i32; 10] =
        [90 as i32, 64 as i32, 20 as i32,
         1 as i32, 52 as i32, 35 as i32,
         40 as i32, 62 as i32, 61 as i32,
         41 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
